src/gt4py/next/program_processors/runners/dace_fieldview/transformations/loop_blocking.py

```diff
@@ -0,0 +1,607 @@
+        blocking_size: Optional[int] = None,
+        blocking_parameter: Optional[Union[gtx_common.Dimension, str]] = None,
+    ) -> None:
+        super().__init__()
+        if isinstance(blocking_parameter, gtx_common.Dimension):
+            blocking_parameter = gtx_dace_fieldview_util.get_map_variable(blocking_parameter)
+        if blocking_parameter is not None:
+            self.blocking_parameter = blocking_parameter
+        if blocking_size is not None:
+            self.blocking_size = blocking_size
+
+    @classmethod
+    def expressions(cls) -> Any:
+        return [dace.sdfg.utils.node_path_graph(cls.outer_entry)]
+
+    def can_be_applied(
+        self,
+        graph: Union[dace.SDFGState, dace.SDFG],
+        expr_index: int,
+        sdfg: dace.SDFG,
+        permissive: bool = False,
+    ) -> bool:
+        """Test if the map can be blocked.
+
+        The test involves:
+        - The map must be at the top level.
+        - The block dimension must be present (exact string match).
+        - The map range must have step size of 1.
+        - The partition must exists (see `partition_map_output()`).
+        """
+        if self.blocking_parameter is None:
+            raise ValueError("The blocking dimension was not specified.")
+        elif self.blocking_size is None:
+            raise ValueError("The blocking size was not specified.")
+
+        outer_entry: dace_nodes.MapEntry = self.outer_entry
+        map_params: list[str] = outer_entry.map.params
+        map_range: dace_subsets.Range = outer_entry.map.range
+        block_var: str = self.blocking_parameter
+
+        scope = graph.scope_dict()
+        if scope[outer_entry] is not None:
+            return False
+        if block_var not in outer_entry.map.params:
+            return False
+        if map_range[map_params.index(block_var)][2] != 1:
+            return False
+        if self.partition_map_output(graph, sdfg) is None:
+            return False
+
+        return True
+
+    def apply(
+        self,
+        graph: Union[dace.SDFGState, dace.SDFG],
+        sdfg: dace.SDFG,
+    ) -> None:
+        """Creates a blocking map.
+
+        Performs the operation described in the doc string.
+        """
+
+        # Now compute the partitions of the nodes.
+        independent_nodes, dependent_nodes = self.partition_map_output(graph, sdfg)  # type: ignore[misc]  # Guaranteed to be not `None`.
+
+        # Modify the outer map and create the inner map.
+        (outer_entry, outer_exit), (inner_entry, inner_exit) = self._prepare_inner_outer_maps(graph)
+
+        # Reconnect the edges
+        self._rewire_map_scope(
+            outer_entry=outer_entry,
+            outer_exit=outer_exit,
+            inner_entry=inner_entry,
+            inner_exit=inner_exit,
+            independent_nodes=independent_nodes,
+            dependent_nodes=dependent_nodes,
+            state=graph,
+            sdfg=sdfg,
+        )
+
+    def _rewire_map_scope(
```

Comment:
If this function doesn't need to be a method because it doesn't use `self`, it should be declared as a module-level function outside of the class. Check here:  https://google.github.io/styleguide/pyguide.html#217-function-and-method-decorators
 > Never use staticmethod unless forced to in order to integrate with an API defined in an existing library. Write a module-level function instead.

Reply:
You are right it does not _need_ to be a method. However, making it a free function feels wrong to me, as the functionality is now split, as some parts are implemented in the class and one (super important one) is a free function. For me this does not make any sense.
For me a class group stuff together and now you tear it apart. It would be different, if the free function could potentially be used somewhere else, but the function is super specific.